sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -246,11 +246,11 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
 
       inferredSchema match {
         case Some(dataSchema) =>
-          val schema = StructType(dataSchema ++ relation.tableMeta.partitionSchema)
           if (inferenceMode == INFER_AND_SAVE) {
-            updateCatalogSchema(relation.tableMeta.identifier, schema)
+            updateDataSchema(relation.tableMeta.identifier, dataSchema)
```

Comment:
Hi, @cloud-fan . 
SPARK-22306 also reports the unexpected change of table `owner`. To resolve SPARK-22306, we need to keep the original `owner` during `INFER_AND_SAVE`. It seems to be the only remaining thing here. Could you consider that in this PR please, too?

Reply:
owner is in table properties?